source/includes/steps-fle-convert-to-a-remote-master-key-aws.yaml

```diff
@@ -15,6 +15,42 @@ content: |
      `Adding permissions to a user <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_change-permissions.html>`__
      to set these permissions.
 
+  .. important:: Use IAM roles
+
+     When deploying your CSFLE-enabled application to a production environment,
+     authenticate your application through an IAM role instead of an IAM user.
+
+     To authenticate with an IAM role, specify your temporary IAM role credentials
+     in your KMS provider object as follows:
+
+     .. code-block:: json
+
+        {
+          "accessKeyId":"<temporary access key ID>",
+          "secretAccessKey":"<temporary secret access key>",
+          "sessionToken":"<session token>"
+        }
+     
+     You can get your temporary IAM role credentials through the following mechanisms:
+
+     - Call ``AssumeRole``
+     - `Retrieve Credentials from EC2 Instance Metadata <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials>`__
```

Comment:
[question] is it worth explicitly mentioning/ linking to how to update expired temporary credentials?

Reply:
Adding content explaining that users must implement logic to handle updating expired temporary credentials.

Context: I think all readers need to be aware of is "MongoDB does not refresh your credentials for you." Actually providing an implementation of how to refresh credentials I think goes too far into AWS world (I think in this case that would be providing code snippets that [request a secure token ](https://docs.aws.amazon.com/code-samples/latest/catalog/javav2-sts-src-main-java-com-example-sts-AssumeRole.java.html)on a 403 error).

If Kenn White or Cynthia want this level of detail we can add it, but I'm not sure it makes sense to do at this time, espcecially that next-gen FLE will handle this automatically for users.

Reply:
That makes sense, sounds good. Thanks for the context